output/out_logzio.go

```diff
@@ -112,51 +138,64 @@ func FLBPluginFlush(data unsafe.Pointer, length C.int, tag *C.char) int {
 			break
 		}
 
-		log, err := serializeRecord(ts, C.GoString(tag), record)
+		log, err := serializeRecord(ts, C.GoString(tag), record, outputs[id].ltype, id)
 		if err != nil {
 			continue
 		}
-		plugin.Send(log)
+		plugin.Send(log, outputs[id].client)
 	}
 
-	return plugin.Flush()
+	return plugin.Flush(outputs[id].client)
 }
 
 //FLBPluginExit When Fluent Bit will stop using the instance of the plugin,
 // it will trigger the exit callback.
 //export FLBPluginExit
 func FLBPluginExit() int {
-	plugin.Flush()
+	plugin.Flush(nil)
 	return output.FLB_OK
 }
 
 func initConfigParams(ctx unsafe.Pointer) error {
-	debug, err := strconv.ParseBool(plugin.Environment(ctx, "logzio_debug"))
+	debug, err := strconv.ParseBool(output.FLBPluginConfigKey(ctx, "logzio_debug"))
 	if err != nil {
 		debug = false
 	}
 
-	logger = NewLogger(outputName, debug)
-	logger.Debug("initializing output plugin..")
+	outputId := output.FLBPluginConfigKey(ctx, "id")
+
+	if outputs == nil {
+		outputs = make(map[string]LogzioOutput)
+	}
+
+	if outputId == "" {
```

Comment:
Is it mandatory to set id? what happens if someone not set it ( for backward compatibility )